### ISSUE TITLE
Support go versioning

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#18480]
### What would you like Renovate to be able to do?

Create a new `go` versioning and use that for `gomod` manager instead of `npm`. Use `bump` as the default rangeStrategy for dependencies.

### If you have any ideas on how this should be implemented, please tell us here.

It seems simple, but might be challenging.

First of all, everything in a `go.mod` file is essentially a range. A value like `v1.0.0` is the equivalent of `^v1.0.0` in `npm`. `isSingleVersion()` should always return `false`.

However when we fetch from a `go` datasource, for that we need to treat results like versions (e.g. sort `v1.0.0`, `v1.0.1`, etc).

In other words `v1.0.0` is both a version and a range.. and I'm not sure if our existing logic can handle that.

 A value is valid if it starts with `v` and is otherwise a valid `semver` version.

### Is this a feature you are interested in implementing yourself?

Maybe
